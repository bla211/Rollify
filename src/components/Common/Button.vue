<template>
  <button class="submit-btn" @click="action(data, method)" :class="colorClass" :disabled="disabled">
    <slot>
      <span class="btn-title">{{title}}</span>
    </slot>
  </button>
</template>

<script>
export default {
  props: {
    title: String,
    data: Object,
    action: Function,
    colorClass: String,
    disabled: Boolean,
    method: String
  }
}
</script>


<style scoped lang="scss">
 @import "@/styles/app.scss";

 %submit-btn-shared{
   @include appearance(none);

   font-family: $font-family;
   color: $grey-9;
   font-weight: 400;
   text-align: center;
   width: 100%;
   padding: 10px;
   float: left;
   margin-top: 6px;
   border: none; 
   cursor: pointer;
   font-size: 1.6rem;
   &:disabled{
     opacity: .5;
     cursor: not-allowed;
   }
 }

 .green_btn{
   @extend %submit-btn-shared;
   background: $informa-green;
 }
 .light_blue_btn{
   @extend %submit-btn-shared;
   background: $light-blue;
 }
 .med_blue_btn{
   @extend %submit-btn-shared;
   background: $med-blue;
 }

 .absolute_bottom{
   position: absolute;
   bottom: 0;
   left: 0;
 }

 .small{
   width: 200px;
 }
</style>
