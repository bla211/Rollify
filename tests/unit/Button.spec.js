import { shallowMount } from '@vue/test-utils'
import Button from '@/components/Common/Button.vue'

describe('Button.vue', () => {
  it('calls action when button is clicked', () => {
    const action = jest.fn();
    const wrapper = shallowMount(Button, {
      propsData: {
        action
      }
    });
    const btn = wrapper.find('.submit-btn');
    btn.trigger('click');
    expect(action).toHaveBeenCalled();
  });

it('renders html properly when passed props', () => {
    const title = 'Button Title';
    const colorClass = 'green_btn';
    const wrapper = shallowMount(Button, {
      propsData: {
        colorClass, title
      }
    });
    const btn = wrapper.find('.submit-btn');
    expect(btn).toMatchSnapshot();
  })
});
