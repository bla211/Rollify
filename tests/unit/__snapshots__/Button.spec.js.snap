// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue renders html properly when passed props 1`] = `<div class="submit-btn green_btn"><span class="btn-title">Button Title</span></div>`;
